Solutions/Marain.Workflow.Abstractions/Marain/Workflows/WorkflowEngine.cs

```diff
@@ -182,6 +198,9 @@ private Task ProcessInstanceWithLeaseAsync(IWorkflowTrigger trigger, string inst
                         await this.workflowInstanceStore.UpsertWorkflowInstanceAsync(instance, workflowInstancePartitionKey).ConfigureAwait(false);
                         await this.InitializeInstanceAsync(instance, workflow, context).ConfigureAwait(false);
                         await this.workflowInstanceStore.UpsertWorkflowInstanceAsync(instance, workflowInstancePartitionKey).ConfigureAwait(false);
+
+                        // We only record this final version post-initialization in the change log.
+                        await this.workflowInstanceChangeLog.RecordWorkflowInstanceChangeAsync(null, instance, workflowInstancePartitionKey).ConfigureAwait(false);
```

Comment:
As above, why not persist the change log entry first?

Reply:
See the answer above.